Makefile

```diff
@@ -1,16 +1,17 @@
-BIN := index-provider
+BIN_SUBDIR := cmd/provider
 
 .PHONY: all build clean test
 
-all: build
+all: vet test build
 
-build: $(BIN)
+build: 
+	cd $(BIN_SUBDIR) && go build
 
 docker: Dockerfile clean
 	docker build . --force-rm -f Dockerfile -t indexer-reference-provider:$(shell git rev-parse --short HEAD)
 
-$(BIN): vet test
-	cd cmd/provider && go build -o ../../$(@)
+install:
+	cd $(BIN_SUBDIR) && go install
```

Comment:
I generally prefer build of a binary over `go install` which can often put the binary into an unexpected location.

Reply:
There is still a `build` target.  The biggest difference is that it does not move the executable command into the repo root directory, but keeps it in the cmd/provider directory.  This was done to provide what is generally more expected behavior.  The `install` target is simply convenience.